sdks/go/test/integration/primitives/windowinto.go

```diff
@@ -91,3 +91,32 @@ func WindowSums_GBK(s beam.Scope) {
 func WindowSums_Lifted(s beam.Scope) {
 	WindowSums(s.Scope("Lifted"), stats.SumPerKey)
 }
+
+// TriggerWindowSums, much like WindowSums described above has an addition of configuring
+// a trigger here. SetDefault works fine. Other triggers such as SetAlways throws
+// pane decoding error.
+func TriggerWindowSums(s beam.Scope, sumPerKey func(beam.Scope, beam.PCollection) beam.PCollection) {
+	timestampedData := beam.ParDo(s, &createTimestampedData{Data: []int{4, 9, 2, 3, 5, 7, 8, 1, 6}}, beam.Impulse(s))
+
+	windowSize := 3 * time.Second
+
+	validate := func(s beam.Scope, wfn *window.Fn, in beam.PCollection, expected ...interface{}) {
+		// Window the data.
+		windowed := beam.WindowInto(s, wfn, in, beam.WindowTrigger{Name: window.Always})
+		// To get the pane decoding error, change above statement to
+		// windowed := beam.WindowInto(s, wfn, in, beam.WindowTrigger{Name: window.Always})
+		// Perform the appropriate sum operation.
+		sums := sumPerKey(s, windowed)
+		// Drop back to Global windows, and drop the key otherwise passert.Equals doesn't work.
+		sums = beam.WindowInto(s, window.NewGlobalWindows(), sums)
+		sums = beam.DropKey(s, sums)
+		passert.Equals(s, sums, expected...)
+	}
+
+	// Use fixed windows to divide the data into 3 chunks.
+	validate(s.Scope("Fixed"), window.NewFixedWindows(windowSize), timestampedData, 15, 15, 15)
```

Comment:
Doesn't the "Always" trigger happen on every element? Should we be actually expecting the sums, or the individual elements?  (I don't know what's correct here, outside of the default trigger.)

Reply:
Yes. I think there is something missing in the trigger implementation. Will be working on it next.